server/src/main/scala/org/http4s/server/package.scala

```diff
@@ -1,13 +1,44 @@
 package org.http4s
 
-import scalaz.{OptionT, Kleisli}
+import scalaz.{Equal, Kleisli, Monoid}
 import scalaz.concurrent.Task
-import scalaz.std.option._
 import scalaz.syntax.kleisli._
-import scalaz.syntax.monad._
-import scalaz.syntax.traverse._
+
+package server {
+
+  /**
+    * Encapsulates the notion of fallthrough orElse for a Service
+    * For any given B, if a Fallthrough[B] exists within implicit context
+    * then [[Service#orElse]] can be used.
+    */
+  trait Fallthrough[B] {
```

Comment:
This may be the old Java programmer in me talking, and totally doesn't matter, but I see Fallthrough and companion in its own file.